crates/turborepo-api-client/src/telemetry.rs

```diff
@@ -0,0 +1,46 @@
+use async_trait::async_trait;
+use reqwest::Method;
+use serde::Serialize;
+
+use crate::{retry, AnonAPIClient, Error};
+
+#[async_trait]
+pub trait TelemetryClient {
+    async fn record_telemetry<T>(
+        &self,
+        events: Vec<T>,
+        telemetry_id: &str,
+        session_id: &str,
+    ) -> Result<(), Error>
+    where
+        T: Serialize + std::marker::Send;
+}
+
+#[async_trait]
+impl TelemetryClient for AnonAPIClient {
+    async fn record_telemetry<T>(
+        &self,
+        events: Vec<T>,
+        telemetry_id: &str,
+        session_id: &str,
+    ) -> Result<(), Error>
+    where
+        T: Serialize + std::marker::Send,
```

Comment:
So we originally had this typing for analytics, but it felt rather opaque since you could pass any serializable + Send type. Could we use a concrete telemetry type? If circular dependencies are an issue, the type definitions can go into `turborepo-vercel-api`

Reply:
Yea I tried that initially just to avoid the generic here, but it ends up with circular crate dependencies in that case (even with `turborepo-vercel-api`, because then the it has to depend on `turborepo-telemetry` to pull the event types in). 

I don't mind the generic here because everything is locked down on the backend, and it also creates a single place to define everything related to event data for anyone to check to audit what we're doing. 

But if there's a better way to make it work lmk and we can get it in.

Reply:
Why can't the event types be in `turborepo-vercel-api`? That's what I ended up doing for analytics

Reply:
Let me see if I can decouple them from the builders a bit to make it work.